tensorflow_model_optimization/python/core/clustering/keras/cluster_integration_test.py

```diff
@@ -91,42 +129,95 @@ def testValuesRemainClusteredAfterTraining(self):
     self.assertLessEqual(len(unique_weights), self.params["number_of_clusters"])
 
   @keras_parameterized.run_all_keras_modes(always_skip_v1=True)
-  def testEndToEnd(self):
-    """Test End to End clustering."""
+  def testEndToEndSequential(self):
```

Comment:
Nit. Consider it as a note for a potential future cleanup PR. 

To make unit tests easier to read and iterate on, it's generally good to minimize the time it takes to understand what a unit test is actually testing. At the moment, someone has to go down three function calls
(testEndToEndSequential -> self.end_to_end_testing -> self.verify_tflite) to get all the information. 

https://github.com/tensorflow/model-optimization/blob/master/tensorflow_model_optimization/python/core/sparsity/keras/prune_integration_test.py#L176 has some examples to reference (e.g. testPrunesZeroSparsity_IsNoOp) that I think do a somewhat better job in that area. 

Ways of improvement (with the above as examples) could mean:
1) Clearer naming of unit tests to indicate expected behavior - testEndToEndSequential lets us know we are testing a Sequential model but not what's being tested (which is under `clusters_check` and `verify_tflite`).
2) Have more of the most important parts of the flow directly under `testEndToEndSequential` instead of nesting it in multiple functions, even if it means some degree of redundancy. The important parts in my opinion are:
- We have a Sequential model (the actual layers don't matter much, as long as they have trainable weights.
- We train the model (the parameters in the calls to `compile` and `fit` don't matter - technically how it's trained doesn't matter either in this test, so in `testPrunesZeroSparsity_IsNoOp`, this is hidden in a _train_model helper function).
- We strip the model for clustering.
- We do the two checks
3) Clearer function names: each of the two checks could be in a helper function to make the flow easier to understand, but it's ideal that it's clear what the checks are doing. What does `clusters_check` and `_verify_tflite` mean? And even if we're instantiating an instance of the TFLite interpreter, what does that actually guarantee (e.g. if the goal is to guarantee there is a runnable model, shouldn't the converter always create a valid one or shouldn't there be other ways of checking that other than creating an interpreter (maybe there isn't)).